x-pack/plugins/observability_solution/synthetics/server/synthetics_service/authentication/check_has_privilege.ts

```diff
@@ -9,18 +9,20 @@ import { SecurityIndexPrivilege } from '@elastic/elasticsearch/lib/api/types';
 import { UptimeEsClient } from '../../lib';
 import { SyntheticsServerSetup } from '../../types';
 import { getFakeKibanaRequest } from '../utils/fake_kibana_request';
-import { serviceApiKeyPrivileges, syntheticsIndex } from '../get_api_key';
+import { getServiceApiKeyPrivileges, syntheticsIndex } from '../get_api_key';
 
 export const checkHasPrivileges = async (
   server: SyntheticsServerSetup,
   apiKey: { id: string; apiKey: string }
 ) => {
+  const { isServerless } = server;
+  const { indices: index, cluster } = getServiceApiKeyPrivileges(isServerless);
   return await server.coreStart.elasticsearch.client
```

Comment:
It's not related to your changes but this `await` is not necessary neither the `async` declaration. As the `server.coreStart.elasticsearch.client` returns a promise the checkHasPrivileges client is the one that needs to await the response.

Reply:
Totally agree, wish Kibana had a rule on unneeded `async`/`await`. I have this lint rule in the recorder and it makes the code a lot cleaner IMO.

I didn't notice it as I wasn't paying attention to the code around the bits I needed to modify.

I've also updated the function below this as it suffers from the same unneeded sugar.

635854b96805f29452cabc0a7dcb2a9a5194f6dc